lib/ash_json_api/json_schema/json_schema.ex

```diff
@@ -559,9 +559,10 @@ defmodule AshJsonApi.JsonSchema do
             "format" => "date-time"
           }
 
-        {:array, _type} ->
+        {:array, type} ->
           %{
-            "type" => "any"
+            "type" => "array",
+            "items" => attribute_filter_schema(type)
```

Comment:
So this isn't really how filters work. Providing an actual schema here would be difficult but *potentially* doable. Specifically, we support a whole range of operators/functions that work on specific types. The acceptable input for `filter` has evolved *tons* since the original inception of this schema. We should chat separately about how to make it better.

But for this specifically, filtering a list of items isn't done by accepting a list of items of the same type.

Reply:
Ah OK, thanks. I'll revert this change.

Turns out swagger doesn't allow the `any` type, giving me this schema validation error:

```
Structural error at paths./json_api/guides/for-version.get.parameters.2.schema.type
should be equal to one of the allowed values allowedValues: array, boolean, integer, number, object, string
```

I'll have a deeper look into how complex filters work in AshJsonApi.

Reply:
Swagger not supporting an `any` type seems really tough. Like we can't necessarily spec out everything all the time (like custom types with no type swagger schema implementation).